oak_runtime/src/tls.rs

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use std::{
+    fs::File,
+    io::{self, BufReader},
+    path::Path,
+    sync::Arc,
+};
+use tokio_rustls::rustls::{
+    internal::pemfile::{certs, rsa_private_keys},
+    Certificate, NoClientAuth, PrivateKey, ServerConfig,
+};
+
+/// Represents TLS identity to use for HTTP server pseudo-nodes.
+#[derive(Default, Clone)]
+pub struct TlsConfig {
+    certs: Vec<Certificate>,
+    keys: Vec<PrivateKey>,
+}
+
+impl TlsConfig {
+    pub fn new(cert_path: &str, key_path: &str) -> Option<Self> {
+        let certs = match load_certs(&Path::new(cert_path)) {
+            Ok(certs) => certs,
+            Err(_) => return None,
+        };
+
+        let keys = match load_keys(&Path::new(key_path)) {
+            Ok(keys) => keys,
+            Err(_) => return None,
+        };
+        Some(TlsConfig { certs, keys })
+    }
+}
+
+pub(crate) fn to_server_config(tls_config: TlsConfig) -> Arc<ServerConfig> {
+    let mut cfg = ServerConfig::new(NoClientAuth::new());
+    // Select a certificate to use.
+    let private_key = tls_config.keys[0].clone();
+    let _ = cfg
```

Comment:
nit: I think it would be clearer to use something like if `let Err(error) = cfg...` to do error logging rather than doing the logging in the map_err (which is not really supposed to have side effects) and then ignoring the result.